cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -295,10 +298,33 @@ synchronized void checkActiveUserTasks() {
     }
   }
 
+  private synchronized void removeFromPurgatory(UserTaskInfo userTaskInfo) {
```

Comment:
I am wondering whether there is a reason why we  keeps submitted items in purgatory until the task expires in `UserTaskManager`. looks to me it is enough that purgatory only maintains item in `Pending` and `Approved` status, once an item it discarded or submitted we can directly recycle it in purgatory. 

Currently there are 2 place saves the information about approved request information, `purgatory` and `UserTaskManager`. If we want to know what tasks are approved, we can directly go to `USER_TASK` endpoint since we know if and only if a POST request is approved, it will be executed and we can get more information there(like execution result).

Reply:
I though a task can only be in UserTaskManager when it is approved?

Reply:
Thanks for bringing up these points -- let me clarify:

Submitted tasks are in purgatory to ensure auditability, abstraction, and to ensure that if the request has already been submitted, the user is not attempting to create another user task with the same parameters and endpoint [added this sanity check in the latest commit]. The auditability lets us know details on an already executed request  -- e.g. who submitted the request and when. The abstraction helps us filter out access to UserTasks in purgatory-level.